File: app/src/at-hoangtran/java/com/asiantech/intern20summer1/intro/DivingDeeper.java
package com.asiantech.intern20summer1.intro;

import java.util.ArrayList;

public class DivingDeeper {
    public static void main(String[] args) {
        int[] intArray = {2,5,4,8,6,3,4,8};
        int k = 2;
        String s = "adf5asd";
        System.out.println(extractEachKth(intArray, k));
        System.out.println(firstDigit(s));
        System.out.println(differentSymbolsNaive(s));
        System.out.println(arrayMaxConsecutiveSum(intArray,k));
    }
    static int[] extractEachKth(int[] inputArray, int k) {
//        Given array of integers, remove each kth element from it.
        int l = inputArray.length;
        ArrayList<Integer> a = new ArrayList<>();
        for(int i =0;i<l;i++){
            if((i+1)%k!=0){
                a.add(inputArray[i]);
            }
        }
        Object[] bo = new Object[a.size()];
        int[] bi = new int[a.size()];
        bo=a.toArray();
        for(int i = 0;i <bi.length;i++){
            bi[i] = (Integer)bo[i];
        }
        return bi;
    }
    static char firstDigit(String inputString) {
//        Find the leftmost digit that occurs in a given string.
        char c = ' ';
        int l = inputString.length();
        for(int i = 0;i<l;i++){
            if(inputString.charAt(i)>=48&&inputString.charAt(i)<=57){
                c =inputString.charAt(i);
                break;
            }
        }
        return c;
    }
    static int differentSymbolsNaive(String s) {
//        Given a string, find the number of different characters in it.
        int l = s.length();
        ArrayList<Character>c = new ArrayList<Character>();
        for(int i = 0;i<l;i++){
            int count = 0;
            for(int j =0;j<c.size();j++){
                if(s.charAt(i)==c.get(j)){
                    count++;
                }
            }
            if(count<1){
                c.add(s.charAt(i));
            }
        }
        return c.size();
    }
    static int arrayMaxConsecutiveSum(int[] inputArray, int k) {
//        Given array of integers, find the maximal possible sum of some of its k consecutive elements.
        int l = inputArray.length;
        int max = 0;
        for(int i =0;i<=l-k;i++){
            int sum = 0;
            for(int j = 0;j<k;j++){
                sum += inputArray[i+j];
            }
            if(sum>max){
                max = sum;
            }
        }
        return max;
    }

}

File: app/src/at-hoangtran/java/com/asiantech/intern20summer1/kotlinIntro/DivingDeeper.kt
package com.asiantech.intern20summer1.kotlinIntro

class DivingDeeper {
    fun extractEachKth(inputArray: MutableList<Int>, k: Int): MutableList<Int> {
//        Given array of integers, remove each kth element from it.
        var a = mutableListOf<Int>()
        for (i in inputArray.indices) {
            if ((i + 1) % k != 0) {
                a.add(inputArray[i])
            }
        }
        return a
    }

    fun firstDigit(inputString: String): Char {
//        Find the leftmost digit that occurs in a given string.
        var c = ' '
        for (i in inputString.indices) {
            if (inputString[i].isDigit()) {
                c = inputString[i]
                break
            }
        }
        return c
    }

    fun differentSymbolsNaive(s: String): Int {
//        Given a string, find the number of different characters in it.
        var tmp = mutableListOf<Char>()
        var count = 0
        for (i in s.indices) {
            if (!tmp.contains(s[i])) {
                tmp.add(s[i])
                count++
            }
        }
        return count
    }

    fun arrayMaxConsecutiveSum(inputArray: MutableList<Int>, k: Int): Int {
//        Given array of integers, find the maximal possible sum of some of its k consecutive elements.
        var max = 0
        for (i in 0..inputArray.size - k) {
            var sum: Int = 0
            for (j in 0..k - 1) {
                sum += inputArray[i + j]
            }
            if (max < sum) {
                max = sum
            }
        }
        return max
    }

}

fun main(args: Array<String>) {
    var nah = DivingDeeper()
    val intArray = mutableListOf(2, 4, 5, 6, 8, 23, 6, 7, 875, 2)
    println(nah.extractEachKth((intArray), 6))
    println(nah.firstDigit("fgdg1"))
    println(nah.differentSymbolsNaive("dfgdfhcb"))
    println(nah.arrayMaxConsecutiveSum(intArray, 2))
}
